Standards/scs-01xx-v1-dns.md

```diff
@@ -0,0 +1,170 @@
+For internal connectivity and discoverability between cloud resources, DNS is used to make virtual machines addressable via their name within tenant networks.
+DNS can also be used to publish DNS records for virtual machines that have external connectivity by integrating and using the OpenStack DNS v2 API, e.g. by integrating the Designate service.
+
+## Terminology
+
+| Term | Meaning |
+|---|---|
+| CSP | Cloud Service Provider, provider managing the OpenStack infrastructure |
+| DNS | Domain Name System |
+| DNS recursor | Recursive DNS resolver |
+| network | OpenStack Neutron network resource unless stated otherwise |
+| OVN | Virtual networking component of the Open vSwitch project and a mechanism driver choice for OpenStack Neutron as an alternative to the OVS driver |
+| OVS | Abbreviation for Open vSwitch, a virtualized switch solution and mechanism driver choice for OpenStack Neutron |
+| port | OpenStack Neutron port resource unless stated otherwise |
+| SCS | Sovereign Cloud Stack |
+| VM | Virtual Machine, also known as "server" resource in OpenStack Nova |
+
+## Motivation
+
+The behavior and feature set regarding DNS functionalities can differ greatly between OpenStack infrastructures, depending on their individual configuration.
+
+With this standard the SCS project aims to establish a baseline for reliable and consistent DNS features in SCS cloud environments.
+
+## Design Considerations
+
+OpenStack offers a lot of extension choices and even a dedicated service (Designate) for DNS functionality.
+The standard should make sure that a specified level of DNS functionality can be reached while taking into account that not all settings and choices might be feasible for CSPs to implement.
+
+### Options considered
+
+#### Making Designate mandatory
+
+To offer a consistent feature set to customers, the SCS project could consider to make Designate mandatory in a sense that SCS-compliant clouds would need to integrate the service, make it available to customers and properly configure it for publishing DNS records.
+This would offer easy DNS-as-a-Service functionality to customers.
+
+However, this would also require solid DNS expertise at CSP-side to properly set up and integrate Designate and DNS zones as Designate does not act as a full DNS server on its own but instead relies on external DNS providers or self-hosted DNS infrastructures that the CSP needs to integrate into it.
+
+#### Mandating the use of DNSSEC
+
+The DNSSEC extension to DNS ensures authenticity and integrity of the data provided to DNS resolvers.
+This protects against attacks like cache poisoning and increases the trustworthiness of the DNS infrastructure.
+
+This is a direct improvement for customers and yields low complexity for the CSP to implement since it is supported by most major DNS implementations natively.
```

Comment:
Thanks for sharing your view on this. This is very valuable!

Do you consider the challenges with DNSSEC usage big enough to demote this to a recommendation?
Currently the standard 1)  makes DNSSEC mandatory for any local recursors and 2) mandates the use of public DNS servers that offer DNSSEC if no local recursor is used.

Reply:
@artificial-intelligence This text does not say you should run a DNSSEC auth.

This text only says that you should deploy a DNSSEC recursor, which AFAIK has none of the organisational complexities you mention.

Reply:
It still has, of course, a lot of inert local complexity, e.g. the CSP will be the first point of contact for any customer who has dnssec related problems and must be able to deal with them.

That means you need trained support representatives which can debug and detect dnssec failures, e.g. when customers misconfigure their external domains because they don't know better.

Also there are still some subtle implementation bugs, e.g. some resolvers don't really properly support dnssec or only support it since $recent_version.

So I'm not really against mandating dnssec, you know, but I would like to see an honest analysis what the up- and downsides are and imho the current analysis is very superficially done on only a high level imho.

For example:
DNSSEC needs the availability of TCP Port 53 (strictly speaking you also need this for non-dnssec supporting resolvers, but it's way easier to get away with problems in your TCP based DNS setup if you don't do dnssec).

The default (forwarding) dns resolver in ml2/ovs is dnsmasq, which is not designed for large networks, it has no ability to configure it's available connection limit for TCP based dns replies. If you want to raise this, you have to recompile dnsmasq.

Further limitations apply to dnsmasq as well:
https://github.com/liquidm/dnsmasq/blob/master/FAQ#L31

Reply:
I'll resolve this after todays discussion in the IaaS Breakout room, as the complexity of dnssec is only concentrated on a resolver provided by the CSP in this case, which enables customers of the CSP to work with dnssec.